src/timeline/widgets/ImageItem.cc

```diff
@@ -83,7 +84,14 @@ ImageItem::ImageItem(QSharedPointer<MatrixClient> client,
         url_                 = QString("%1/_matrix/media/r0/download/%2")
                  .arg(client_.data()->getHomeServer().toString(), media_params);
 
-        setImage(QPixmap(filename));
+        if (!data.isNull()) {
+                data->reset();
```

Comment:
The return code of this shouldn't be ignored unless you can prove that it's guaranteed to succeed.